ingestion/src/metadata/ingestion/source/database/databricks/connection.py

```diff
@@ -91,4 +93,11 @@ def custom_executor(engine, statement):
         ),
     ]
 
-    return test_connection_db_common(engine, steps)
+    timeout_seconds = service_connection.timeOut
+    timer = threading.Timer(timeout_seconds, lambda: None)
+    timer.start()
+    result = test_connection_engine(engine, steps)
+    timer.cancel()
```

Comment:
not sure I like this approach, I believe we can leave it clean at the top `test_connection_db_common` implementation

Reply:
I think we can reuse the `timeout` definition we currently have, instead of replicating the same logic somewhere else, if we define the functions as:

```python
def test_connection_db_common(connection: Engine, timeout_seconds: int = 120, steps=None) -> TestConnectionResult:
    """
    Default implementation is the engine to test.
    Test that we can connect to the source using the given engine
    :param connection: Engine to test
    :return: None or raise an exception if we cannot connect
    """
    return timeout(timeout_seconds)(test_connection_engine)(connection, steps)
```